### PR TITLE
[Common actions] Add Copy To Clipboard recipe

### PR DESCRIPTION
**Pull request by aaronccasanova**
### WHY are these changes introduced?

Closes https://github.com/Shopify/polaris-internal/issues/1521

### WHAT is this pull request doing?

This PR introduces a minimal Copy to Clipboard recipe following the [Common Actions guidance](https://polaris.shopify.com/patterns/common-actions/overview). Additionally, a new `useCopyToClipboard` hook was added to facilitate the native copy behavior and state management for the copied timeout.

https://github.com/Shopify/polaris/assets/32409546/17110be3-557b-4a0f-b7c1-a1d2eab57c2e

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#install-dependencies-and-build-workspaces)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

### 🎩 checklist

- [ ] Tested a [snapshot](https://github.com/Shopify/polaris/blob/main/documentation/Releasing.md#-snapshot-releases)
- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

### CODE REVIEW

**Review by alex-page — APPROVED** [review]

**Comment by sarahill** [comment]
@aaronccasanova  It's feeling a little flashy even when I wait the full transition time for the initial "Copy" tooltip. There's a flash the first couple times and then it goes into a flicker the next couple (see gif).

**Review by sam-b-rose — APPROVED** [review]
📋 💯

File: polaris-react/src/components/Tooltip/Tooltip.tsx [hunk 1]
```diff
@@ -158,7 +158,13 @@ export function Tooltip({
...
     (event: React.KeyboardEvent) => {
-      if (event.key !== 'Escape') return;
+      if (
+        event.key !== 'Escape' &&
+        event.key !== 'Enter' &&
+        event.key !== ' '
+      ) {
+        return;
+      }
       handleClose?.();
```
Comment (aaronccasanova) [inline polaris-react/src/components/Tooltip/Tooltip.tsx]: Updates the `Tooltip` to dismiss on enter/space key interactions (matching the behavior `onClick`)

https://github.com/Shopify/polaris/assets/32409546/4c576622-3bc8-43b4-9ec9-f876cf98d7c1

File: polaris-react/src/components/Tooltip/Tooltip.tsx [hunk 2]
```diff
@@ -206,6 +212,11 @@ export function Tooltip({
         handleFocus();
       }}
       onBlur={() => {
+        if (hoverDelayTimeout.current) {
+          clearTimeout(hoverDelayTimeout.current);
+          hoverDelayTimeout.current = null;
+        }
```
Comment (aaronccasanova) [inline polaris-react/src/components/Tooltip/Tooltip.tsx:217]: Prevents the `Tooltip` from flickering/reopening when clicking the activator before the `hoverDelayTimeout` (video at 25% speed)

https://github.com/Shopify/polaris/assets/32409546/497a4b99-e8ee-4330-83ab-de632d36e897

**Review by sarahill — APPROVED** [review]
This looks good and feels pretty smooth and easy to interact with 👏

**Review by chloerice — APPROVED** [review]
:shipit:

**Review by sam-b-rose — APPROVED** [review]
🚢

File: .changeset/flat-eagles-press.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/polaris': minor
+---
+
+Added `useCopyToClipboard` hook for building copy actions matching common actions guidelines. Updated `Tooltip` to dismiss on enter/space key interactions.
```
Comment (aaronccasanova) [inline .changeset/flat-eagles-press.md]:
```suggestion
Added `useCopyToClipboard` hook for building copy actions matching common actions guidelines. Updated `Tooltip` to dismiss on child enter/space key interactions.
```

**Pull request merged**